identity/src/androidTest/java/com/android/identity/IdentityCredentialTest.java

```diff
@@ -0,0 +1,172 @@
+    checkStaticAuthData(1, 100000); // 1 key x 100 KBytes
+  }
+
+  @LargeTest
+  @Test
+  public void storeStaticAuthenticationDataLarge()
+          throws IdentityCredentialException {
+    Assume.assumeTrue("Test fails on API <= 28", VERSION.SDK_INT >= 29 /* Android 10+ */);
+    // Per b/234563696, I've confirmed these failing on:
+    // Pixel 2 API 24
+    // Pixel 2 API 26
+    // Pixel 3 API 28
+    checkStaticAuthData(10, 10000); // fails
+    checkStaticAuthData(100, 1000); // flaky
+    checkStaticAuthData(100, 10000); // fails
+  }
+
+  private void checkStaticAuthData(int numAuthKeys, int staticAuthDataSizeBytes)
+          throws IdentityCredentialException {
+    final int usesPerKey = 3;
+    ProvisioningTest.createCredential(store, CREDENTIAL_NAME);
+    try {
+      IdentityCredential credential = Preconditions.checkNotNull(getCredentialIfPresent());
+
+      storeFakeStaticAuthData(credential, usesPerKey, numAuthKeys, staticAuthDataSizeBytes);
+
+      // Check that this doesn't throw (it did throw in internal b/234563696).
+      try {
+        getCredentialIfPresent();
+      } catch (RuntimeException e) {
+        String msg = String.format(Locale.US,
+                "Failed to load credential for %d authKeys with %d bytes static auth data each",
+                numAuthKeys, staticAuthDataSizeBytes);
+        throw new IdentityCredentialException(msg, e);
+      }
+    } finally {
+      deleteCredentialFromStoreIfPresent();
+    }
+  }
+
+  @Before
+  public void setUp() {
+    deleteCredentialFromStoreIfPresent(); // clean up in case a previous test didn't
+  }
+
+  @After
+  public void tearDown() {
+    deleteCredentialFromStoreIfPresent();
+    store = null;
+  }
+
+  private static void storeFakeStaticAuthData(IdentityCredential credential,
+          int maxUsesPerKey, int numKeys, int staticAuthDataSizeBytes) {
+    Random random = new Random(31337 + numKeys);
+    credential.setAvailableAuthenticationKeys(numKeys, maxUsesPerKey);
+    Collection<X509Certificate> authKeys = credential.getAuthKeysNeedingCertification();
+    assertEquals(numKeys, authKeys.size());
+
+    for (X509Certificate authKey : authKeys) {
+      byte[] staticAuthData = new byte[staticAuthDataSizeBytes];
+      random.nextBytes(staticAuthData);
+      Calendar expirationDate = Calendar.getInstance();
+      expirationDate.add(Calendar.YEAR, 1);
+      try {
+        credential.storeStaticAuthenticationData(authKey, expirationDate, staticAuthData);
+      } catch (UnknownAuthenticationKeyException e) {
+        throw new AssertionFailedError("Auth authKey not recognized: " + e);
+      }
+    }
+  }
+
+  private @Nullable IdentityCredential getCredentialIfPresent() {
+    try {
+      return store.getCredentialByName(CREDENTIAL_NAME, CIPHER_SUITE);
+    } catch (CipherSuiteNotSupportedException e) {
+      throw new AssertionFailedError("Unexpected exception: " + e);
+    }
+  }
+
+  private void deleteCredentialFromStoreIfPresent() {
```

Comment:
I don't see any value for including this wrapper method. Let's just call `deleteCredentialByName` directly.

Reply:
Done. The value originally was that:

-  Instead of just extracting the CREDENTIAL_NAME constant, we extracted the whole piece of logic (which feels more meaningful than the constant)
- It provided a single place where to comment why we're calling a deprecated method rather than following the (unreliable) replacement that is suggested in the deprecation annotation. But now that it's only called in one place, it makes sense to inline it.
 
Side note: IMO introducing the delete() method on IdentityCredential was probably an API mistake, not only because it makes it harder to get out of a corrupt state, but also because I feel that IdentityCredential should have been immutable and not offer any API to interact with the database / storage.